src/models/ReportDashboardWidget.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+namespace sadi01\bidashboard\models;
+
+use sadi01\bidashboard\helpers\CoreHelper;
+use sadi01\bidashboard\traits\AjaxValidationTrait;
+use sadi01\bidashboard\traits\CoreTrait;
+use yii\base\Widget;
+use yii\behaviors\BlameableBehavior;
+use yii\behaviors\TimestampBehavior;
+use yii\db\ActiveRecord;
+use yii\helpers\ArrayHelper;
+use yii2tech\ar\softdelete\SoftDeleteBehavior;
+use Yii;
+
+
+/**
+ * This is the model class for table "report_dashboard_widget".
+ *
+ * @property int $id
+ * @property int $slave_id
+ * @property int $dashboard_id
+ * @property int $widget_id
+ * @property int $display_type
+ * @property int|null $chart_type
+ * @property int $status
+ * @property int widget_field_format
+ * @property int $created_at
+ * @property int $updated_at
+ * @property int $deleted_at
+ * @property int|null $updated_by
+ * @property int|null $created_by
+ *
+ * @property ReportDashboard $dashboard
+ * @property ReportWidget $widget
+ */
+class ReportDashboardWidget extends ActiveRecord
+{
+    use AjaxValidationTrait;
+    use CoreTrait;
+
+    const STATUS_DELETED = 0;
+    const STATUS_ACTIVE = 1;
+    const STATUS_INACTIVE = 2;
+
+    const DISPLAY_CHART = 1;
+    const DISPLAY_TABLE = 2;
+    const DISPLAY_CARD = 3;
+
+    const CHART_LINE = 1;
+    const CHART_COLUMN = 2;
+    const CHART_PIE = 3;
+    const CHART_AREA = 4;
+    const CHART_WORD_CLOUD = 5;
+
+    const FORMAT_NUMBER = 1;
+    const FORMAT_CURRENCY = 2;
+
+    const SCENARIO_CREATE = 'create';
+    const SCENARIO_UPDATE = 'update';
+
+    const VALID = 1;
+    const IN_VALID = 0;
+
+    public $title;
+    public $description;
+    public $rangeType;
+    public $isValid;
+    public $rangeDateCount;
+    public $results = [];
+
+    public $cardResultCount;
+
+    public static function getDb()
+    {
+        return Yii::$app->biDB;
+    }
+
+    public static function tableName()
+    {
+        return '{{%report_dashboard_widget}}';
+    }
+
+    public function rules()
+    {
+        return [
+            [['slave_id'], 'default', 'value' => function () {
+                return Yii::$app->params['bi_slave_id'] ?? null;
+            }],
+            /*[['widget_id', 'display_type', 'widget_field'], 'required', 'on' => self::SCENARIO_CREATE],
+            [['widget_id', 'display_type', 'widget_field', 'widget_field_format'], 'required', 'on' => self::SCENARIO_UPDATE],*/
+            [['widget_id', 'display_type', 'widget_field', 'widget_field_format'], 'required'],
+            ['chart_type', 'required', 'when' => function($model) {
```

Comment:
top of all rules

Reply:
ReportDashboardWidget.php completely removed!